### PR TITLE
NDJson import and export

### PR DESCRIPTION
**Pull request by awaismslm**
In this PR, I have implemented NDJson import and export.
I have used Golang's encode and decode functions for NDJson.

### CODE REVIEW

File: cmd_export.go [hunk 1]
```diff
@@ -184,35 +178,13 @@ func cmdExport() error {
...
-					subnet.String(), data,
-				)
-			} else {
-				fmt.Fprintf(
-					outFile, ",\n    \"%s\": %s",
-					subnet.String(), data,
-				)
-			}
+			record["range"] = subnet.String()
+			enc := json.NewEncoder(outFile)
```
Comment (UmanShahzad) [inline cmd_export.go:182]: We shouldn't create a new encoder on each iteration.

File: cmd_import.go [hunk 2]
```diff
@@ -463,27 +463,51 @@ func cmdImport() error {
...
 		}
 
-		firstSeen := false
-		var data map[string]map[string]string
-		json.Unmarshal([]byte(file), &data)
-		for networkStr, recordMap := range data {
-			// on first object?
-			if !firstSeen {
-				firstSeen = true
+		dataStream := json.NewDecoder(strings.NewReader(string(file)))
```
Comment (UmanShahzad) [inline cmd_import.go:466]: The implementation isn't solving things properly and not bringing the benefit of NDJSON in the first place; we need to stream. See the NOTE above - that note needs to be removed by fixing that issue.

**Review by UmanShahzad — COMMENTED** [review]
Hey @awaismslm can you also show examples of how an actual import/export looks? Wanna see the interfaces we can expect from the resulting file in either direction.

File: cmd_import.go (hunk 2 above)
Comment (awaismslm) [reply cmd_import.go:466]: working on it now. Streaming was not mentioned in the task, so I did not do it. I'll try to finish it by Monday IA.

Comment (awaismslm) [reply cmd_import.go]: @UmanShahzad I was working on the said thing and found that [json.Decoder](https://godoc.org/encoding/json#Decoder) has a built-in streaming feature. :) I have modified the code a bit. Now we are not loading the entire file in memory. `Decoder` takes the file path as input and treats the file as a stream. Please take a look.

Please note that the `Decoder's` official documentation does not have the explicit mention of streaming from the file but it does have mentions of input stream which I am using as the file path. Although I did find many Stackoverflow's answers which mention this feature, [here is an answer relevant to the context.](https://stackoverflow.com/questions/34388083/read-entire-file-of-newline-delimited-json-blobs-to-memory-and-unmarshal-each-bl)

**Comment by awaismslm** [comment]
> Hey @awaismslm can you also show examples of how an actual import/export looks? Wanna see the interfaces we can expect from the resulting file in either direction.

for import, I have used the [asn.json](https://storage.googleapis.com/ipinfoio/artifacts/v1/asn.json.gz) as input and got the mmdb file as output. 

for export, I have used the mmdb file generated from import as input and got json file as output which is ndjosn.
here is the sample of the output json file .

I can share the files over Gdrive with you if you want to take a look at the complete files.

`{"asn":"AS13335","country":"US","domain":"cloudflare.com","name":"Cloudflare, Inc.","network":"1.0.0.0-1.0.0.255","range":"1.0.0.0/24","type":"hosting"}
{"asn":"AS38803","country":"AU","domain":"gtelecom.com.au","name":"Wirefreebroadband Pty Ltd","network":"1.0.4.0-1.0.7.255","range":"1.0.4.0/22","type":"isp"}
{"asn":"AS2519","country":"JP","domain":"arteria-net.com","name":"ARTERIA Networks Corporation","network":"1.0.16.0-1.0.16.255","range":"1.0.16.0/24","type":"isp"}
{"asn":"AS141748","country":"PH","domain":"qdatacom.com","name":"QUANTUM DATA COMMUNICATIONS, INC.","network":"1.0.32.0-1.0.32.255","range":"1.0.32.0/24","type":"isp"}
{"asn":"AS18144","country":"JP","domain":"enecom.co.jp","name":"Energia Communications,Inc.","network":"1.0.64.0-1.0.127.255","range":"1.0.64.0/18","type":"business"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.0.128.0-1.0.255.255","range":"1.0.128.0/17","type":"isp"}
{"asn":"AS13335","country":"US","domain":"cloudflare.com","name":"Cloudflare, Inc.","network":"1.1.1.0-1.1.1.255","range":"1.1.1.0/24","type":"hosting"}
{"asn":"AS138421","country":"CN","domain":"chinaunicom.cn","name":"China Unicom","network":"1.1.8.0-1.1.8.255","range":"1.1.8.0/24","type":"hosting"}
{"asn":"AS2519","country":"JP","domain":"arteria-net.com","name":"ARTERIA Networks Corporation","network":"1.1.64.0-1.1.126.255","range":"1.1.64.0/19","type":"isp"}
{"asn":"AS2519","country":"JP","domain":"arteria-net.com","name":"ARTERIA Networks Corporation","network":"1.1.64.0-1.1.126.255","range":"1.1.96.0/20","type":"isp"}
{"asn":"AS2519","country":"JP","domain":"arteria-net.com","name":"ARTERIA Networks Corporation","network":"1.1.64.0-1.1.126.255","range":"1.1.112.0/21","type":"isp"}
{"asn":"AS2519","country":"JP","domain":"arteria-net.com","name":"ARTERIA Networks Corporation","network":"1.1.64.0-1.1.126.255","range":"1.1.120.0/22","type":"isp"}
{"asn":"AS2519","country":"JP","domain":"arteria-net.com","name":"ARTERIA Networks Corporation","network":"1.1.64.0-1.1.126.255","range":"1.1.124.0/23","type":"isp"}
{"asn":"AS2519","country":"JP","domain":"arteria-net.com","name":"ARTERIA Networks Corporation","network":"1.1.64.0-1.1.126.255","range":"1.1.126.0/24","type":"isp"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.1.128.0-1.1.255.255","range":"1.1.128.0/17","type":"isp"}
{"asn":"AS24406","country":"CN","domain":"cnnic.cn","name":"China Internet Network Infomation Center","network":"1.2.4.0-1.2.4.255","range":"1.2.4.0/24","type":"business"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.2.128.0-1.2.165.255","range":"1.2.128.0/19","type":"isp"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.2.128.0-1.2.165.255","range":"1.2.160.0/22","type":"isp"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.2.128.0-1.2.165.255","range":"1.2.164.0/23","type":"isp"}
{"asn":"AS9737","country":"TH","domain":"tot.co.th","name":"TOT Public Company Limited","network":"1.2.166.0-1.2.166.255","range":"1.2.166.0/24","type":"business"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.2.167.0-1.2.255.255","range":"1.2.167.0/24","type":"isp"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.2.167.0-1.2.255.255","range":"1.2.168.0/21","type":"isp"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.2.167.0-1.2.255.255","range":"1.2.176.0/20","type":"isp"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.2.167.0-1.2.255.255","range":"1.2.192.0/18","type":"isp"}
{"asn":"AS23969","country":"TH","domain":"ntplc.co.th","name":"TOT Public Company Limited","network":"1.4.128.0-1.4.255.255","range":"1.4.128.0/17","type":"isp"}
{"asn":"AS4725","country":"JP","domain":"softbank.jp","name":"SoftBank Corp.","network":"1.5.0.0-1.5.255.255","range":"1.5.0.0/16","type":"business"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.0.0-1.6.135.255","range":"1.6.0.0/17","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.0.0-1.6.135.255","range":"1.6.128.0/21","type":"hosting"}
{"asn":"AS132215","country":"IN","domain":"powergrid.in","name":"Power Grid Corporation of India Limited","network":"1.6.136.0-1.6.136.255","range":"1.6.136.0/24","type":"isp"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.137.0-1.6.218.255","range":"1.6.137.0/24","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.137.0-1.6.218.255","range":"1.6.138.0/23","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.137.0-1.6.218.255","range":"1.6.140.0/22","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.137.0-1.6.218.255","range":"1.6.144.0/20","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.137.0-1.6.218.255","range":"1.6.160.0/19","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.137.0-1.6.218.255","range":"1.6.192.0/20","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.137.0-1.6.218.255","range":"1.6.208.0/21","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.137.0-1.6.218.255","range":"1.6.216.0/23","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.137.0-1.6.218.255","range":"1.6.218.0/24","type":"hosting"}
{"asn":"AS137130","country":"IN","domain":"pnbindia.com","name":"Punjab National Bank","network":"1.6.219.0-1.6.219.255","range":"1.6.219.0/24","type":"business"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.220.0-1.6.225.255","range":"1.6.220.0/22","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.220.0-1.6.225.255","range":"1.6.224.0/23","type":"hosting"}
{"asn":"AS132215","country":"IN","domain":"powergrid.in","name":"Power Grid Corporation of India Limited","network":"1.6.226.0-1.6.228.255","range":"1.6.226.0/23","type":"isp"}
{"asn":"AS132215","country":"IN","domain":"powergrid.in","name":"Power Grid Corporation of India Limited","network":"1.6.226.0-1.6.228.255","range":"1.6.228.0/24","type":"isp"}
{"asn":"AS4755","country":"IN","domain":"tatacommunications.com","name":"TATA Communications formerly VSNL is Leading ISP","network":"1.6.229.0-1.6.230.255","range":"1.6.229.0/24","type":"isp"}
{"asn":"AS4755","country":"IN","domain":"tatacommunications.com","name":"TATA Communications formerly VSNL is Leading ISP","network":"1.6.229.0-1.6.230.255","range":"1.6.230.0/24","type":"isp"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.231.0-1.6.255.255","range":"1.6.231.0/24","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.231.0-1.6.255.255","range":"1.6.232.0/21","type":"hosting"}
{"asn":"AS9583","country":"IN","domain":"sifycorp.com","name":"Sify Limited","network":"1.6.231.0-1.6.255.255","range":"1.6.240.0/20","type":"hosting"}`

File: cmd_import.go (hunk 2 above)
Comment (UmanShahzad) [reply cmd_import.go]: That's cool!

**Review by UmanShahzad — APPROVED** [review]
Will release this separately once we put this into the `ipinfo` cli too.

**Pull request merged**